src/Tribe/Main.php

```diff
@@ -760,12 +760,15 @@ public function post_types() {
 			tribe_update_option( 'ticket-enabled-post-types', $defaults );
 		}
 
+		// Remove WooCommerce Product and EDD post types to prevent recursion fatal error on save.
+		$filtered_post_types = array_diff( $options['ticket-enabled-post-types'], [ 'product', 'download' ] );
```

Comment:
Hey just a thought ... 

It caught my eye that we do an explicit cast to array on line 721. Should we that here, too? 

```suggestion
		$filtered_post_types = array_diff( (array) $options['ticket-enabled-post-types'], [ 'product', 'download' ] );
```

1. We could then drop the explicit cast on line 771
2. It would avoid a PHP warning if `$options['ticket-enabled-post-types']` is null _(unlikely_ for sure, but possible if a user deselects all post types from the relevant setting)